src/AspNetCoreModuleV2/InProcessRequestHandler/dllmain.cxx

```diff
@@ -90,46 +93,27 @@ CreateApplication(
     _Out_ IAPPLICATION      **ppApplication
 )
 {
-    HRESULT                 hr = S_OK;
-    IN_PROCESS_APPLICATION *pApplication = NULL;
-    REQUESTHANDLER_CONFIG  *pConfig = NULL;
-
-    // Initialze some global variables here
     InitializeGlobalConfiguration(pServer);
 
     try
     {
-        hr = REQUESTHANDLER_CONFIG::CreateRequestHandlerConfig(pServer, pHttpApplication, &pConfig);
-        if (FAILED(hr))
-        {
-            goto Finished;
-        }
+        REQUESTHANDLER_CONFIG  *pConfig = NULL;
+        RETURN_IF_FAILED(REQUESTHANDLER_CONFIG::CreateRequestHandlerConfig(pServer, pHttpApplication, &pConfig));
 
-        pApplication = new IN_PROCESS_APPLICATION(pServer, pConfig);
+        BOOL disableStartupPage = pConfig->QueryDisableStartUpErrorPage();
 
-        pConfig = NULL;
+        auto pApplication = std::make_unique<IN_PROCESS_APPLICATION>(pServer, pConfig);
 
-        *ppApplication = pApplication;
-    }
-    catch (std::bad_alloc&)
-    {
-        hr = E_OUTOFMEMORY;
-    }
-
-Finished:
-    if (FAILED(hr))
-    {
-        if (pApplication != NULL)
+        if (FAILED(pApplication->LoadManagedApplication()))
         {
-            delete pApplication;
-            pApplication = NULL;
+            *ppApplication = new StartupExceptionApplication(pServer, disableStartupPage);
```

Comment:
```
pApplication  = make_unique<StartupExceptionApplication>();
```

and then `*ppApplication = pApplication.release();` after if

Reply:
Can't do that exactly as pApplication is of type `std::unique_ptr<InProcessApplication>`. Would you prefer I don't call new?

Reply:
Need to call `LoadManagedApplication()` on the InProcessApplication.